R/final_size.R

```diff
@@ -1,125 +1,225 @@
-#' If a vector, must be the same length as the demography vector, otherwise the
-#' vector will be recycled. Default value is 0.001 for all groups.
-#' @param prop_suscep  Proportion of each group susceptible. Null assumption is
-#' fully susceptible
+#' @title Final size of an epidemic
+#'
+#' @description `final_size` calculates the final size of an epidemic outbreak
+#' in a population with heterogeneous mixing, and with heterogeneous
+#' susceptibility to infection such as that conferred by an immunisation
+#' programme.
+#'
+#' @details
+#' # Solver options
+#'
+#' The `control` argument accepts a list of solver options, with the iterative
+#' solver taking two extra arguments than the Newton solver. This is an optional
+#' argument, and default options are used within the solver functions if an
+#' argument is missing. Arguments provided override the solver defaults.
+#'
+#' ## Common options
+#'
+#' 1. `iterations`: The number of iterations over which to solve for the final
+#' size, unless the error is below the solver tolerance. Default = 10000.
+#' 2. `tolerance`: The solver tolerance; solving for final size ends when the
+#' error drops below this tolerance. Defaults to set `1e-6`.
+#'
+#' ## Iterative solver options
+#' 1. `step_rate`: The solver step rate. Defaults to 1.9 as a value found to
+#' work well.
+#' 2. `adapt_step`: Boolean, whether the solver step rate should be changed
+#' based on the solver error. Defaults to TRUE.
+#'
+#' @param r0 The basic reproductive number \eqn{R_0} of the disease.
+#' @param contact_matrix Social contact matrix. Entry \eqn{m_{ij}} gives average
+#' number of contacts in group \eqn{i} reported by participants in group \eqn{j}
+#' @param demography_vector Demography vector. Entry \eqn{v_{i}} gives
+#' proportion of total population in group \eqn{i} (model will normalise
+#' if needed).
+#' @param p_susceptibility A matrix giving the probability that an individual
+#' in demography group \eqn{i} is in risk (or susceptibility) group \eqn{j}.
+#' Each row represents the overall distribution of individuals in demographic
+#' group $i$ across risk groups, and each row *must sum to 1.0*.
+#' @param susceptibility A matrix giving the susceptibility of individuals in
+#' demographic group \eqn{i} and risk group \eqn{j}.
+#' @param solver Which solver to use. Options are "iterative" (default) or
+#' "newton", for the iterative solver, or the Newton solver, respectively.
+#' Special conditions apply when using the Newton solver, see the `control`
+#' argument.
+#' @param control A list of named solver options, see *Solver options*.
+#'
 #' @keywords epidemic model
+#' @return A data.frame of the proportion of infected individuals, within each
+#' demography group and susceptibility group combination.
+#' If the demography groups and susceptibility groups are named, these
+#' names are added to relevant columns. If the groups are not named, synthetic
+#' names are added (e.g. `demo_grp_1`, `susc_grp_1`).
```

Comment:
The first 3 columns of your output are your `susceptibility` matrix in tidy format. Would it make sense to get it in this format as input already?

Reply:
I don't have a preference, happy to go either way. Possibly a topic for another PR